pennylane/_grad.py

```diff
@@ -181,3 +182,206 @@ def _jacobian_function(*args, **kwargs):
+        to the arguments in ``argnum``
+    """
+
+    if argnum > len(args) - 1:
+        raise ValueError(
+            "The value of 'argnum' has to be between 0 and {}; got {}".format(len(args) - 1, argnum)
+        )
+
+    x = _np.array(args[argnum])
+    gradient = _np.zeros_like(x, dtype="O")
+
+    if x.ndim == 0 and idx is not None:
+        raise ValueError(
+            "Argument {} is not an array, 'idx' should be set to 'None';"
+            " got {}".format(argnum, idx)
+        )
+
+    if idx is None:
+        idx = list(_np.ndindex(*x.shape))
+    else:
+        bounds = _np.array(x.shape) - 1
+        for _idx in idx:
+
+            if (_np.array(_idx) > bounds).any():
+                raise ValueError(
+                    "Indices in 'idx' can not be greater than {}; got {}".format(
+                        tuple(bounds), _idx
+                    )
+                )
+
+    for i in idx:
+        shift = _np.zeros_like(x)
+        shift[i] += 0.5 * delta
+        gradient[i] = (
+            f(*args[:argnum], x + shift, *args[argnum + 1 :], **kwargs)
+            - f(*args[:argnum], x - shift, *args[argnum + 1 :], **kwargs)
+        ) * delta ** -1
+
+    return gradient
+
+
+def _fd_second_order_centered(f, argnum, delta, *args, idx=None, **kwargs):
+    r"""Uses a central finite difference approximation to compute the second-order
+    derivative :math:`\frac{\partial^2 f(x)}{\partial x_i \partial x_j}` of the function ``f``
+    with respect to the argument ``argnum``.
+
+    Args:
+        f (function): function with signature ``f(*args, **kwargs)``
+        argnum (int): the argument with respect to which the gradient is taken
+        delta (float): step size used to evaluate the finite difference
+        idx (list[int]): if argument ``args[argnum]`` is an array, `idx`` specifies
+            the indices ``i, j`` of the arguments to differentiate
+
+    Returns:
+        (float or array): the second-order derivative of the input function with respect
+        to the arguments in ``argnum``
+    """
+
+    if argnum > len(args) - 1:
+        raise ValueError(
+            "The value of 'argnum' has to be between 0 and {}; got {}".format(len(args) - 1, argnum)
+        )
+
+    x = _np.array(args[argnum])
+
+    if x.ndim == 0 and idx is not None:
+        raise ValueError(
+            "Argument {} is not an array, 'idx' should be set to 'None';"
+            " got {}".format(argnum, idx)
+        )
+
+    if idx is None:
+        idx = [(), ()]
+    else:
+        bounds = _np.array(x.shape) - 1
+        if len(idx) > 2:
+            raise ValueError(
+                "The number of indices given in 'idx' can not be greater than two; got {} indices".format(
+                    len(idx)
+                )
+            )
+
+        if (_np.array(idx) > bounds).any():
+            raise ValueError(
+                "Indices in 'idx' can not be greater than {}; got {}".format(tuple(bounds), idx)
+            )
```

Comment:
Similar as above, I recommend removing the `else` and simply letting NumPy exceptions catch this.